File: Java Banking System/demo/src/main/java/com/example/TFSA.java
package com.example;
import java.util.Scanner;

public class TFSA extends SavingsAccount{

    public int goal;
    public int contributions;
    public TFSA(int accountNumber, String accountHolderName, int balance, int minimumBalance, float interestRate,
            int withdrawLimit, int goal) {
        super(accountNumber, accountHolderName, balance, minimumBalance, interestRate, withdrawLimit);
        this.goal = goal;
        this.contributions = 0;
    }
    
    @Override
    public void deposit(int amount) {
        this.contributions += amount;
        if (this.contributions >= this.goal) {
            System.out.println("Financial goal reached");
            Scanner scanner = new Scanner(System.in);
            System.out.println("Please enter a new financial goal to work towards (> 0): ");
            String newGoal = scanner.nextLine();
            try {
                Integer.parseInt(newGoal);
                this.goal = Integer.parseInt(newGoal);
                this.contributions = 0;
                super.deposit(amount);
            }
            catch (NumberFormatException e) {
                System.out.println("Invalid input");
            }
            scanner.close();
        }
    }

    @Override
    public void displayAccountDetails(){
        System.out.println("Displaying information...");
        System.out.println("Account Number: " + this.getAccountNumber());
        System.out.println("Account Name: " + this.getAccountHolderName());
        System.out.println("Account Balance: " + this.getBalance());
        System.out.println("Minimum Balance: " + this.getMinimumBalance());
        System.out.println("Interest Rate: " + this.getInterestRate());
        System.out.println("Withdraw Limit: " + this.getWithdrawLimit());
        System.out.println("Progress towards goal: " + this.contributions + "/" + this.goal + "\n");
    }
}

File: Python Banking System/TFSA.py
from SavingsAccount import SavingsAccount
class TFSA(SavingsAccount):
    def __init__(self, accountNumber, accountHolderName, balance, minimumBalance, interestRate, withdrawLimit, goal):
        super().__init__(accountNumber, accountHolderName, balance, minimumBalance, interestRate, withdrawLimit)
        self._goal = goal
        self._contributions = 0

    def deposit(self, amount):
        self._contributions += amount
        if self._contributions >= self._goal:
            print("Financial goal reached")
            newGoal = input("Please enter a new financial goal to work towards (> 0): ")
            if newGoal.isnumeric() and int(newGoal) > 0:
                self._goal = int(newGoal)
                self._contributions = 0
            else:
                while not newGoal.isnumeric() or int(newGoal) <= 0:
                    newGoal = input("Invalid input, please enter a new financial goal to work towards (> 0): ")
                self._goal = int(newGoal)
                self._contributions = 0
        return super().deposit(amount)

    def displayAccountDetails(self):
        print("Displaying information...")
        print(f"Account Number: {self._accountNumber}")
        print(f"Account Name: {self._accountHolderName}")
        print(f"Account Balance: {self._balance}")
        print(f"Minimum Balance: {self._minimumBalance}")
        print(f"Interest Rate: {self._interestRate}")
        print(f"Withdraw Limit: {self._withdrawLimit}")
        print(f"Progress towards goal: {self._contributions}/{self._goal}")
        print()